sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -246,6 +248,29 @@ class Analyzer(
       CleanupAliases)
   )
 
+  /**
+   * Replace interval with startEnd + interval and substitute child plan with DateTimeOverlaps.
+   */
+  object ResolveDateTimeOverlaps extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
+      case p: LogicalPlan => p.transformAllExpressions {
+        case u @ UnresolvedDateTimeOverlaps(leftStart, leftEnd, _, _)
+          if leftEnd.dataType.isInstanceOf[CalendarIntervalType] =>
+          u.copy(leftEnd = Add(leftStart, leftEnd))
```

Comment:
Why did you do nothing for right here? We cannot replace it with `DateTimeOverlaps` in this match?

Reply:
SGTM. I’ll refine this